src/core/exception_hacks.cc

```diff
@@ -76,6 +76,12 @@ static dl_iterate_fn dl_iterate_phdr_org() {
 
 static std::vector<dl_phdr_info> phdrs_cache;
 
+// The destruction order in a single translation unit is stable
```

Comment:
Can I haz more info, pls?  Based on this comment alone, it's not clear who/when depends on `phdrs_cache_valid` being set to the correct value.  It's also not clear why the [construct-on-first-use idiom](https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2) isn't leveraged here, as I would expect.